File: Android_Domaci2/app/src/main/java/com/example/domaci2/MainActivity.java
package com.example.domaci2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.splashscreen.SplashScreen;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SplashScreen splashScreen = SplashScreen.installSplashScreen(this);
        splashScreen.setKeepOnScreenCondition(() -> {
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            checkLogin();
            return false;
        });

        setContentView(R.layout.activity_main);
    }


    private void checkLogin(){
        boolean logged = false;

        SharedPreferences sharedPreferences = getSharedPreferences(getPackageName(), Context.MODE_PRIVATE);
        String username = sharedPreferences.getString(LoginActivity.REG_NAME_KEY,"");

        if(username.equals("")){//nije ulogovan
            Intent intent = new Intent(this, ScreenActivity.class);
            startActivity(intent);
        }
        else{
            Intent intent = new Intent(this, LoginActivity.class);
            startActivity(intent);
        }




    }




}

File: Android_Projekat2/app/src/main/java/com/example/rmaproject2/presentation/view/activities/MainActivity.kt
package com.example.rmaproject2.presentation.view.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.example.rmaproject2.R


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        val splashScreen: SplashScreen = installSplashScreen()
        splashScreen.setKeepOnScreenCondition {

            val sharedPreferences = getSharedPreferences(packageName, MODE_PRIVATE)
            val username = sharedPreferences.getString(LoginActivity.USERNAME, "")
            intent = if (username == "") {
                Intent(this, LoginActivity::class.java)
            } else {
                Intent(this, AppActivity::class.java)
            }
            startActivity(intent)
            finish()
            false
        }

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}